frontend/src/metabase/query_builder/components/DataSelector.jsx

```diff
@@ -661,7 +672,43 @@ export class UnconnectedDataSelector extends Component {
     return null;
   }
 
+  showTableSearch = () => {
+    const { hasTableSearch, steps } = this.props;
+    const { activeStep } = this.state;
+    const hasTableStep = steps.includes(TABLE_STEP);
+    const isAllowedToShowOnActiveStep = [SCHEMA_STEP, DATABASE_STEP].includes(
+      activeStep,
+    );
+
+    return hasTableSearch && hasTableStep && isAllowedToShowOnActiveStep;
+  };
+
+  handleSearchTextChange = searchText =>
+    this.setState({
+      searchText,
+    });
+
+  handleSearchItemSelect = async item => {
+    const table = convertSearchResultToTableLikeItem(item);
+    await this.props.fetchFields(table.id);
+    if (this.props.setSourceTableFn) {
+      this.props.setSourceTableFn(table.id);
+    }
+    this.popover.current.toggle();
+    this.handleClose();
+  };
+
+  handleClose = () => {
+    this.setState({ searchText: "" });
+  };
+
   render() {
+    const { searchText } = this.state;
+    const { canChangeDatabase, selectedDatabaseId } = this.props;
+    const searchDatabaseId = canChangeDatabase ? null : selectedDatabaseId;
+
+    const isSearchActive = searchText.length >= 2;
```

Comment:
it annoys me terribly that it doesn't search at all on first character. can this be 1? what happens if people have obscure terrible two-letter table names?

Reply:
Personally, I do not see a major difference between the two options, but it would be consistent with the global search to enable search even with 1 character.

@kdoh what do you think?